packages/task/src/browser/task-service.ts

```diff
@@ -77,7 +77,8 @@ export class TaskService implements TaskConfigurationClient {
     @postConstruct()
     protected init(): void {
         // wait for the workspace root to be set
-        this.workspaceService.root.then(async root => {
+        this.workspaceService.onWorkspaceChanged(async event => {
```

Comment:
again, what if we miss an event, `then` is better if you want to get data once for sure, please review other cases where `then` should be used instead of `onWorkspaceChanged`